openedx/adg/admin_dashboard/admin_settings/tests/test_views.py

```diff
@@ -13,75 +13,72 @@
 from openedx.core.djangolib.testing.utils import skip_unless_lms
 
 
-@pytest.mark.parametrize('total_admins, expected_count', [
-    (5, 10),
-    (10, 20),
+@pytest.mark.parametrize('total_admins, user_type, expected_count', [
+    (5, ADG_ADMIN, 10),
+    (10, ADG_ADMIN, 10),
+    (5, BU_ADMIN, 5),
+    (10, BU_ADMIN, 10),
 ])
 @pytest.mark.django_db
 @skip_unless_lms
-def test_admin_accounts_list_queryset(total_admins, expected_count):
+def test_admin_accounts_context_data(
+    total_admins, user_type, expected_count, request_factory
+):
     """
-    Test queryset for `AdminAccountListView` returns expected number of active admins.
+    Test context data for `AdminAccountListView`. Total admins accounts will be `total_admins` + 1 because 1 is
+    attached to request. In case admins are 10+ then it will always returns 10 due to pagination.
     """
-    create_admin_accounts(total_admins, ADG_ADMIN, True)
-    create_admin_accounts(total_admins, BU_ADMIN, True)
-    create_admin_accounts(total_admins, BU_ADMIN, False)
-    assert AdminAccountListView().get_queryset().count() == expected_count
-
-
-@pytest.mark.parametrize('total_admins, expected_count', [
-    (5, 5),
-    (10, 10),
-])
-@pytest.mark.django_db
-@skip_unless_lms
-def test_admin_accounts_context_data(total_admins, expected_count, super_user):
-    """
-    Test context data for `AdminAccountListView`.
-    """
-    create_admin_accounts(total_admins, ADG_ADMIN, True)
+    _, _, adg_admins, _ = create_admin_accounts(total_admins, ADG_ADMIN, True)
+    _, _, bu_admins, _ = create_admin_accounts(total_admins, BU_ADMIN, True)
 
     url = reverse('admin_accounts')
-    admin_accounts_get_request = RequestFactory().get(url)
-    admin_accounts_get_request.user = super_user
+    admin_accounts_get_request = request_factory.get(url)
+
+    admin = adg_admins[0] if user_type == ADG_ADMIN else bu_admins[0]
```

Comment:
Perhaps we should use a constant with descriptive variable name instead of `0`

Reply:
I don't think its really needed for tests. We can go with this.